### ISSUE TITLE
Integrate with React Native Web polyfills

### ISSUE THREAD

**Issue by RangerMauve** [decosoftware/deco-ide#135]
There's been a few efforts to port various React Native APIs to work purely in the browser. It would be super useful if this IDE had the ability to build a project that made use of one or more of these to enable people to create apps that work not only on iOS and Android, but also anywhere that can run a browser.

The libraries that I've seen that seem most appealing are:

[react-web](https://github.com/taobaofed/react-web) Pretty good feature richness, seem to have some own ideas how building should be done. Developers seem to be primarily Chinese so it might have some language barriers if support is needed. Seems to have company backing which is good.
[react-native-web](https://github.com/taobaofed/react-web) Fewer components but the docs are nice and use React Storybook for quick references on how they work, but doesn't seem to have particularly detailed docs. Seems to mostly be run by one person in their spare time. 
[react-native-for-web](https://github.com/KodersLab/react-native-for-web) Seems to be the least polished of the bunch, but could still be a viable option. More text docs than `react-native-web`

Obviously this would be a pretty big undertaking, but if this project had something like this it would be very unique in the ecosystem and could be immensely useful for cross platform development.

**Comment by RangerMauve** [comment]
According to the [react-native-web readme](https://github.com/necolas/react-native-web#why) it seems the project is somehow affiliated with Twitter.

**Comment by dabbott** [comment]
We are going to do this.

We want to make it easier to preview/share RN apps. We were considering going all-in on react-native-web and using it to preview/share your app. For now though, native modules are pretty important to RN, and so we're working on a better way to preview/share that's a little friendlier to native modules. This work includes customizable starter project templates, so it would lend itself nicely to other platforms.

After that we plan on revisiting adding RNW as a third platform.

**Comment by RangerMauve** [comment]
I was about to suggest looking into [https://github.com/dabbott/react-native-web-player](https://github.com/dabbott/react-native-web-player) for previews using web, but then I realized who I was talking to. :sweat_smile:

**Comment by dabbott** [comment]
Hehe :)

I can't wait till the day when we can ditch all this native simulator/xcode/android studio/obj-c/java stuff and go back to the web dev workflow. Seems like we're not *quite* there yet though...

**Comment by RangerMauve** [comment]
Yeah, I'm thinking if react-native-web provided all the same commandline functionality as react-native, it'd help a lot. Right now the hot reload and building is still on users, so it's not as quick to set up.

I guess this issue should remain open for when the project gets time to focus on it?

**Comment by dabbott** [comment]
Yeah, let's just leave it open. Certainly if the react-native packager supported rnw somehow that'd make things dead simple.

**Comment by RangerMauve** [comment]
It seems that react-native's packager was used for `react-vr` bundling files for WebVR. Would it make sense to look into how they did it to re-use it for react-native-web?
